File: Data Structures & Algorithms/reverse-bits/submission-1.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        long long ans=0;
        for(int i = 31; i>=0; --i){
            if((n&1)==1){
                ans += pow(2, i);
            }
            n = n>>1;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/n-queens/submission-1.py
class Solution:
    def solveNQueens(self, n: int) -> List[List[str]]:
        col = set()
        negDiag = set()
        posDiag = set()
        res = [["."] * n for i in range(n)]
        ans = []
        def backtrack(r):
            
            if r == n:
                copy = ["".join(row) for row  in res]
                ans.append(copy)
                return
            for c in range(n):
                if  c in col or (r-c) in negDiag or (r+c) in posDiag:
                    continue
                col.add(c)
                negDiag.add(r-c)
                posDiag.add(r+c)
                res[r][c] = 'Q'
                backtrack(r+1)

                col.remove(c)
                negDiag.remove(r - c)
                posDiag.remove(r+c)
                res[r][c] = '.'
        backtrack(0)
        return  ans